plugins/dominant-color-images/hooks.php

```diff
@@ -134,34 +140,23 @@ function dominant_color_img_tag_add_dominant_color( $filtered_image, string $con
 		return $filtered_image;
 	}
 
-	$data        = '';
-	$extra_class = '';
-
 	if ( ! empty( $image_meta['dominant_color'] ) ) {
-		$data .= sprintf( 'data-dominant-color="%s" ', esc_attr( $image_meta['dominant_color'] ) );
+		$processor->set_attribute( 'data-dominant-color', esc_attr( $image_meta['dominant_color'] ) );
 
-		if ( str_contains( $filtered_image, ' style="' ) ) {
-			$filtered_image = str_replace( ' style="', ' style="--dominant-color: #' . esc_attr( $image_meta['dominant_color'] ) . '; ', $filtered_image );
-		} else {
-			$filtered_image = str_replace( '<img ', '<img style="--dominant-color: #' . esc_attr( $image_meta['dominant_color'] ) . ';" ', $filtered_image );
+		$style_attribute = '--dominant-color: #' . esc_attr( $image_meta['dominant_color'] ) . '; ';
+		if ( null !== $processor->get_attribute( 'style' ) ) {
+			$style_attribute .= $processor->get_attribute( 'style' );
```

Comment:
What if the existing style doesn't end in a semicolon?

Reply:
@westonruter `--dominant-color` CSS is being prepended to ensure it does not break existing styles.

Reply:
Oh, good point 😊